docs/use/formatting_outputs.md

```diff
@@ -0,0 +1,218 @@
+
+import pandas, sys
+print("this is some stdout")
+print("this is some stderr", file=sys.stderr)
+# but what I really want to show is:
+pandas.DataFrame({"column 1": [1, 2, 3]})
+```
+
+(use/format/images)=
+## Images
+
+With the default renderer, for any image types output by the code, we can apply formatting *via* cell metadata.
+The keys should be placed under `myst`, then for the image we can apply all the variables of the standard [image directive](https://docutils.sourceforge.io/docs/ref/rst/directives.html#image):
+
+- **width**: length or percentage (%) of the current line width
+- **height**: length
+- **scale**: integer percentage (the "%" symbol is optional)
+- **align**: "top", "middle", "bottom", "left", "center", or "right"
+- **classes**: space separated strings
+- **alt**: string
+
+Units of length are: 'em', 'ex', 'px', 'in', 'cm', 'mm', 'pt', 'pc'
+
+We can also set a caption (which is rendered as [CommonMark](https://commonmark.org/)) and name, by which to reference the figure:
+
+````md
+```{code-cell} ipython3
+---
+myst:
+  image:
+    width: 200px
+    alt: fun-fish
+    classes: shadow bg-primary
+  figure:
+    caption: |
+      Hey everyone its **party** time!
+    name: fun-fish
+---
+from IPython.display import Image
+Image("images/fun-fish.png")
+```
+````
+
+```{code-cell} ipython3
+---
+myst:
+  image:
+    width: 300px
+    alt: fun-fish
+    classes: shadow bg-primary
+  figure:
+    caption: |
+      Hey everyone its **party** time!
+    name: fun-fish
+---
+from IPython.display import Image
+Image("images/fun-fish.png")
+```
+
+Now we can link to the image from anywhere in our documentation: [swim to the fish](fun-fish)
+
+(use/format/markdown)=
+## Markdown
+
+Markdown output is parsed by MyST-Parser, currently with the configuration set to `myst_commonmark_only=True` (see [MyST configuration options](myst:intro/config-options)).
+
+The parsed Markdown is integrated into the wider documentation, and so it is possible, for example, to include internal references:
+
+```{code-cell} ipython3
+from IPython.display import display, Markdown
+display(Markdown('**_some_ markdown** and an [internal reference](use/format/markdown)!'))
+```
+
+and even internal images can be rendered!
+
+```{code-cell} ipython3
+display(Markdown('![figure](../_static/logo.png)'))
+```
+
+(use/format/ansi)=
+## ANSI Outputs
+
+By default, the standard output/error streams and text/plain MIME outputs may contain ANSI escape sequences to change the text and background colors.
+
+```{code-cell} ipython3
+import sys
+print("BEWARE: \x1b[1;33;41mugly colors\x1b[m!", file=sys.stderr)
+print("AB\x1b[43mCD\x1b[35mEF\x1b[1mGH\x1b[4mIJ\x1b[7m"
+      "KL\x1b[49mMN\x1b[39mOP\x1b[22mQR\x1b[24mST\x1b[27mUV")
+```
+
+This uses the built-in {py:class}`~myst_nb.ansi_lexer.AnsiColorLexer` [pygments lexer](https://pygments.org/).
+You can change the lexer used in the `conf.py`, for example to turn off lexing:
+
+```python
+nb_render_text_lexer = "none"
```

Comment:
@chrisjsewell one last question. Does this option in `conf.py` just control `ansi`?

Reply:
yep, it controls what pygments lexer is applied to plain text outputs (including stdout/stderr)

Reply:
ok - so it does more than just `ansi`. Was just thinking it might be better named `nb_render_text_ansi` but it is more generally a control over the lexer used which contains more than just `ansi` parsing.